notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java

```diff
@@ -765,11 +767,28 @@ public Response updateNoteById(@Parameter(description = "Note id", required = tr
       } else if (!note_.getContent().equals(note.getContent())) {
         if (StringUtils.isBlank(note.getLang())) {
           note_.setContent(note.getContent());
+          note_.setProperties(notePageProperties);
           note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_CONTENT, identity);
         } else {
+          note_.setLang(note.getLang());
           note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_CONTENT, identity);
           note_.setContent(note.getContent());
+          note_.setProperties(notePageProperties);
+        }
+        noteService.createVersionOfNote(note_, identity.getUserId());
+        if (!Utils.ANONYM_IDENTITY.equals(identity.getUserId())) {
+          WikiPageParams noteParams = new WikiPageParams(note_.getWikiType(), note_.getWikiOwner(), newNoteName);
+          noteService.removeDraftOfNote(noteParams, note.getLang());
+        }
+      } else if ((featuredImage != null && (featuredImage.isToDelete() || featuredImage.getUploadId() != null))
+          || !notePageProperties.getSummary().equals(note_.getProperties().getSummary())) {
+        if (StringUtils.isBlank(note.getLang())) {
+          note_.setProperties(notePageProperties);
```

Comment:
IMO the properties should be set only when the properties are really modified

Reply:
and what this condition means ? :  `else if ((featuredImage != null && (featuredImage.isToDelete() || featuredImage.getUploadId() != null))
          || !notePageProperties.getSummary().equals(note_.getProperties().getSummary()))`